### PR TITLE
Kanvi/Upgrade TF to 2.0

### PR DESCRIPTION
**Pull request by kanvi-nervana**
This is an initial PR (Step 1 ) to move towards TF2.0. It upgrades TF version from 1.15.2 to 2.0.0 when the user builds the bridge using the flag --use_tensorflow_2. 
When this flag is used : **./build_ngtf.py --use_tensorflow_2** some changes are applied using a patch **tf2changes.patch** which the user will have to manually revert in case when he wants to go back to using TF 1.15.2 or get a new repo

### CODE REVIEW

**Comment by ashahba** [comment]
@googlebot I censent

File: test/python/test_gathernd.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 import pytest
 
-import tensorflow as tf
+import tensorflow.compat.v1 as tf
```
Comment (adk9) [inline test/python/test_gathernd.py:25]: 👍

File: README.md [hunk 2]
```diff
@@ -176,9 +176,9 @@ Modify the downloaded TensorFlow script to run TensorFlow with nGraph optimizati
 
     import ngraph_bridge
     ...
-    config = tf.ConfigProto()
+    config = tf.compat.v1.ConfigProto()
```
Comment (shresthamalik) [inline README.md:179]: The default is 1.15? will these apis work with 1.15?

Comment (kanvi-nervana) [reply README.md:179]: They are working. All CI pipelines are running with 1.15 and passing.

File: test/ci/buildkite/run_pipeline.sh [hunk 3]
```diff
@@ -7,12 +7,12 @@ echo "BUILDKITE_REPO: $BUILDKITE_REPO"
...
 
 if [[ $PIPELINE_QUEUE = 'cpu' ]]; then
-   TF_PY_WHEEL=tensorflow-1.15.2-cp35-cp35m-linux_x86_64.whl
+   TF_PY_WHEEL=tensorflow-2.0.0-cp36-cp36m-linux_x86_64.whl
    # For the time being - hardcode the file
    # Eventually we will replace the queue and other variables during the pipeline creation
    STEPS_FILE=ngtf-cpu_ubuntu.yaml
 elif [[ $PIPELINE_QUEUE = 'cpu-centos' ]]; then
-   TF_PY_WHEEL=tensorflow-1.15.2-cp36-cp36m-linux_x86_64.whl
+   TF_PY_WHEEL=tensorflow-2.0.0-cp36-cp36m-linux_x86_64.whl
```
Comment (shresthamalik) [inline test/ci/buildkite/run_pipeline.sh:15]: high level, seems like we are always installing 2.0. is this right? Not sure what this file does.

File: test/python/test_mnist_training.py [hunk 4]
```diff
@@ -36,6 +37,8 @@
 
 class TestMnistTraining(NgraphTest):
 
+    #Todo: Unskip this test
+    @pytest.mark.skip(reason="Does not work with tf2.0")
```
Comment (shresthamalik) [inline test/python/test_mnist_training.py:41]: can we disable it only for 2.0

File: test/ci/buildkite/run_pipeline.sh (hunk 3 above)
Comment (kanvi-nervana) [reply test/ci/buildkite/run_pipeline.sh:15]: I am not very sure of this file's purpose either but when you look at the CI pipelines build step, the version printed in the end is 1.15

File: test_ngtf.py [hunk 5]
```diff
@@ -99,7 +99,7 @@ def main():
 
     if (not os.path.isdir(build_dir + '/tensorflow')):
         download_repo(build_dir + "/tensorflow",
-                      "https://github.com/tensorflow/tensorflow.git", "v1.15.2")
+                      "https://github.com/tensorflow/tensorflow.git", "v2.0.0")
```
Comment (shresthamalik) [inline test_ngtf.py:102]: please check this

File: configure_bazel.sh [hunk 6]
```diff
@@ -25,7 +25,7 @@ rm -f .bazelrc
 if python -c "import tensorflow" &> /dev/null; then
     echo 'using installed tensorflow'
 else
-    pip install tensorflow==v1.15.2
+    pip install tensorflow==v2.0.0
```
Comment (shresthamalik) [inline configure_bazel.sh:28]: please check this.

**Pull request merged**